PatelDLTGroupStamps/Views/StampHelper.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  StampHelper.swift
+//  Hiplatelics
+//
+//  Created by patelpra on 6/9/23.
+//
+
+import Foundation
+
+class StampHelper: ObservableObject {
+  @Published var stamps: [Stamp] = [] {
+
+    didSet {
+      Task {
+        try await saveStamps()
+      }
+    }
+  }
+  // MARK: Assignment 1
+  func getStamps() async throws {
+    let (data, response) = try await
+    URLSession.shared.data(from: URL(string: "http://www.grupodlt.com/stamps/hip.json")!)
+
+    guard let response = response as?
+            HTTPURLResponse, response.statusCode == 200 else {
+      throw StampError.invalidResponse
+    }
+
+    Task { @MainActor in
+      self.stamps = try JSONDecoder().decode([Stamp].self, from: data)
+    }
+  }
+
+  // MARK: NEW save method
+  public func saveStamps() async throws {
+    guard let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
+      throw StampError.badURLRequest
+    }
+
+    let jsonFile = documentDirectory.appendingPathComponent("newStamp.json")
+    let urlString = "http://www.grupodlt.com/stamps/hip.json"
```

Comment:
I see a lot of things in this method that could be extracted into individual functions. Try to think in plain English of the things you are doing here:

* getting a URL
* retrieving data from that URL
* getting a URL to write to
* writing data to that location
* handling errors

Having individual functions for each of those actions will give you more testable code.